forte/data/extractor/attribute_extractor.py

```diff
@@ -0,0 +1,189 @@
+
+    @property
+    def state(self) -> Dict:
+        state = super().state
+        state.update({
+            "attribute": self.config.attribute
+        })
+        return state
+
+    @classmethod
+    def from_state(cls, state: Dict) \
+        -> Union["BaseExtractor", "AttributeExtractor"]:
+        obj = BaseExtractor.from_state(state)
+        additional = {
+            "attribute": state["attribute"]
+        }
+        obj.config = Config(obj.config, additional,
+                            allow_new_hparam=True)
+        obj.__class__ = cls
+        return obj
+
+    @staticmethod
+    def get_attribute(entry: Entry, attr: str) -> Any:
+        r"""Functionality: Get the attribute from entry. You can
+        overwrite this function if you have sepcial way to get the
+        attribute from entry.
+
+        Args:
+            entry (Entry): An instance of Entry type, where the
+                attribute will be extracted from.
+            attr (str): The name of the attribute.
+
+        Returns:
+            Any. The attribute extracted from entry.
+        """
+        return getattr(entry, attr)
+
+    @staticmethod
+    def set_attribute(entry: Entry, attr: str, value: Any):
+        r"""Functionality: Set the attribute of an entry to value.
+        You can overwrite this function if you have sepcial way to
+        set the attribute.
+
+        Args:
+            entry (Entry): An instance of Entry type, where the
+                attribute will be set.
+            attr (str): The name of the attribute.
+            value (Any): The value to be set for the attribute.
+        """
+        if attr == "text":
+            raise AttributeError("text attribute of entry cannot "
+                                "be changed.")
+        setattr(entry, attr, value)
+
+    def update_vocab(self, pack: DataPack, instance: Annotation):
+        r"""Functionality: Get all attributes of one instance and
+        add them into the vocabulary.
+
+        Args:
+            pack (DataPack): The datapack that contains the current
+                instance.
+            instance (Annotation): The instance from which the
+                extractor will extractor feature.
+        """
+        for entry in pack.get(self.config.entry_type, instance):
+            element = self.get_attribute(entry, self.config.attribute)
+            if not isinstance(element, Hashable):
+                raise AttributeError("Only hashable element can be"
+                    "added into the vocabulary. Consider setting"
+                    "vocab_method to be raw and do not call update_vocab"
+                    "if you only need the raw attribute value without"
+                    "coverting them into index.")
+            self.add(element)
+
+    def extract(self, pack: DataPack, instance: Annotation) -> Feature:
+        r"""Functionality: Extract attributes of one instance.
+        For example, the text of tokens in one sentence.
```

Comment:
`text` is a special feature we'd better mention this?

Reply:
I think we can only mention it in the `set_attribute`? Because it is `extract` here. Text behaves the same as other attribute.

Reply:
ok, this is not a big deal.